scripts/datacube.py

```diff
@@ -230,61 +174,76 @@ def search_sentinel1(BBOX, catalog, week):
             "args": [
                 {
                     "op": "s_intersects",
-                    "args": [{"property": "geometry"}, geom_BBOX.__geo_interface__],
+                    "args": [{"property": "geometry"}, geom_bbox.__geo_interface__],
                 },
-                {"op": "anyinteracts", "args": [{"property": "datetime"}, week]},
+                {"op": "anyinteracts", "args": [{"property": "datetime"}, date_range]},
                 {"op": "=", "args": [{"property": "collection"}, "sentinel-1-rtc"]},
             ],
         },
     )
     s1_items = search.item_collection()
     print(f"Found {len(s1_items)} Sentinel-1 items")
 
-    s1_gdf = gpd.GeoDataFrame.from_features(s1_items.to_dict())
-    s1_gdf["overlap"] = s1_gdf.intersection(box(*BBOX)).area
-    state = (
-        s1_gdf[["sat:orbit_state", "overlap"]]
-        .groupby(["sat:orbit_state"])
-        .sum()
-        .sort_values(by="overlap", ascending=False)
-        .index[0]
-    )
-    s1_gdf = s1_gdf[s1_gdf["sat:orbit_state"] == state]
-    print("Filtered Sentinel-1 orbit state: ", s1_gdf["sat:orbit_state"].unique())
-    print("Number of scenes filtered by orbit state: ", len(s1_gdf))
+    # Add id as property to persist in gdf
+    for item in s1_items:
+        item.properties["id"] = item.id
+
+    # Try to find enough scenes with same orbit that fully overlap
+    # the S2 bbox.
+    s1_gdf = gpd.GeoDataFrame.from_features(s1_items)
```

Comment:
This will fail if there are no matches. We need a way to handle cases where there are no S1 scenes. Maybe go back to search another date for S2?